```
=========================== short test summary info ============================
FAILED test/test_blocks.py::TestBlocksPostprocessing::test_dataset_is_updated - OSError: Cannot find empty port in range: 7860-7959. You can specify a diff...
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
========= 1 failed, 141 passed, 150 deselected, 63 warnings in 19.95s ==========
```

--- a/pkgs/development/python-modules/gradio-pdf/default.nix
+++ b/pkgs/development/python-modules/gradio-pdf/default.nix
@@ -11,13 +11,13 @@
 
 buildPythonPackage rec {
   pname = "gradio-pdf";
-  version = "0.0.11";
+  version = "0.0.13";
   format = "pyproject";
 
   src = fetchPypi {
     pname = "gradio_pdf";
     inherit version;
-    hash = "sha256-HCfjJdd9DIRqm/dS00xlU9AYGM7U/b3zNo2IgZHtasc=";
+    hash = "sha256-lxfbQSJavJQSYMGqxG7zmg/XT8V8TU2I3zGiq+B/dnw=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/development/python-modules/gradio/client.nix
+++ b/pkgs/development/python-modules/gradio/client.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
   nix-update-script,
@@ -27,7 +28,7 @@
 
 buildPythonPackage rec {
   pname = "gradio-client";
-  version = "1.2.0";
+  version = "1.3.0";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -39,7 +40,7 @@ buildPythonPackage rec {
     # not to be confused with @gradio/client@${version}
     rev = "refs/tags/gradio_client@${version}";
     sparseCheckout = [ "client/python" ];
-    hash = "sha256-l5WHNerSYNXrFGOpAqxxh0JLiFpatxq6a62q83tEavo=";
+    hash = "sha256-UZQWguUN3l0cj2wb2f7A61RTLy9nPYcIEwHIo+F1kR0=";
   };
   prePatch = ''
     cd client/python
@@ -89,6 +90,14 @@ buildPythonPackage rec {
     "test/"
     "-m 'not flaky'"
     #"-x" "-W" "ignore" # uncomment for debugging help
+  ];
+
+  disabledTests = lib.optionals stdenv.isDarwin [
+    # flaky: OSError: Cannot find empty port in range: 7860-7959
+    "test_layout_components_in_output"
+    "test_layout_and_state_components_in_output"
+    "test_upstream_exceptions"
+    "test_httpx_kwargs"
   ];
 
   pythonImportsCheck = [ "gradio_client" ];

--- a/pkgs/development/python-modules/gradio/default.nix
+++ b/pkgs/development/python-modules/gradio/default.nix
@@ -47,6 +47,7 @@
 
   # check
   pytestCheckHook,
+  hypothesis,
   altair,
   boto3,
   gradio-pdf,
@@ -63,7 +64,7 @@
 
 buildPythonPackage rec {
   pname = "gradio";
-  version = "4.40.0";
+  version = "4.41.0";
   format = "pyproject";
 
   disabled = pythonOlder "3.7";
@@ -71,7 +72,7 @@ buildPythonPackage rec {
   # We use the Pypi release, since it provides prebuilt webui assets
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-ChV5E6RfFcOwW6uFqU4/phfHfn27yS8+MKjMVytnlgU=";
+    hash = "sha256-d4li7kQFMzlUVGdm2nTSnj25pTOWIqnZuOvTOtwPLpc=";
   };
 
   # fix packaging.ParserSyntaxError, which can't handle comments
@@ -133,6 +134,7 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     pytestCheckHook
+    hypothesis
     altair
     boto3
     gradio-pdf
@@ -187,6 +189,9 @@ buildPythonPackage rec {
 
     # flaky: OSError: Cannot find empty port in range: 7860-7959
     "test_docs_url"
+    "test_orjson_serialization"
+    "test_dataset_is_updated"
+    "test_multimodal_api"
 
     # tests if pip and other tools are installed
     "test_get_executable_path"
@@ -245,8 +250,8 @@ buildPythonPackage rec {
     "test/test_networking.py"
     # makes pytest freeze 50% of the time
     "test/test_interfaces.py"
-  ] ++ lib.optionals stdenv.isDarwin [
-    # Network-related tests that are flaky on darwin (depend on port availability)
+
+    # Local network tests dependant on port availability (port 7860-7959)
     "test/test_routes.py"
   ];
   pytestFlagsArray = [

